.circleci/config.yml

```diff
@@ -8,108 +8,120 @@ executors:
-            # Must be relative path from root
-            paths:
-                - .gradle
-                - project
+#      # Test
+#      - run:
+#          name: Run tests
+#          command: xvfb-run ./gradlew --info --stacktrace test
+#
+#      - run:
+#          name: Collect test results
+#          command: find . -type f -regex ".*/build/test-results/.*xml" -exec cp -v {} ~/circleci-test-results/junit/ \;
+#      - run:
+#          name: Collect artifacts
+#          command: '[ ! -d build/reports ] || cp -R build/reports ~/circleci-artifacts/'
+#
+#        # Teardown
+#        # Save test results
+#      - store_test_results:
+#          path: ~/circleci-test-results
+#        # Save artifacts
+#      - store_artifacts:
+#          path: ~/circleci-artifacts
+#      - store_artifacts:
+#          path: ~/circleci-test-results
+
+      - run: rm -rf .git
+      # Persist the specified paths (workspace/echo-output) into the workspace for use in build/test jobs.
+      - persist_to_workspace:
+          # Must be an absolute path, or relative path from working_directory. This is a directory on the container which is
+          # taken to be the root directory of the workspace.
+          root: ~/
+          # Must be relative path from root
+          paths:
+            - .gradle
+            - project
 
   deploy_snapshot:
     executor: openjdk_executor
     steps:
-        - attach_workspace:
-            at: ~/
-        - run:
-            name: Deploy snapshot to Artifactory
-            command: ./gradlew -PbuildAccess=opensource -PbuildNumber=$CIRCLE_BUILD_NUM -PbuildTag=$CIRCLE_TAG -PartifactoryUrl=$ARTIFACTORY_URL -PartifactoryRepository=$SNAPSHOT_ARTIFACTORY_REPOSITORY -PartifactoryUsername=$ARTIFACTORY_USERNAME -PartifactoryPassword=$ARTIFACTORY_PASSWORD --info --stacktrace artifactoryPublish
+      - custom_env
+      - attach_workspace:
+          at: ~/
+      - run:
+          name: Deploy snapshot
+          command: ./gradlew --info --stacktrace publishToSonatype
   deploy_release:
     executor: openjdk_executor
     steps:
+      - custom_env
       - attach_workspace:
           at: ~/
       - run:
-          name: Deploy release to Artifactory
-          command: ./gradlew -PbuildAccess=opensource -PbuildNumber=$CIRCLE_BUILD_NUM -PbuildTag=$CIRCLE_TAG -PartifactoryUrl=$ARTIFACTORY_URL -PartifactoryRepository=$RELEASE_ARTIFACTORY_REPOSITORY -PartifactoryUsername=$ARTIFACTORY_USERNAME -PartifactoryPassword=$ARTIFACTORY_PASSWORD --info --stacktrace artifactoryPublish
-      - run:
-          name: Deploy release to Bintray
-          command: ./gradlew -PbuildAccess=opensource -PbuildNumber=$CIRCLE_BUILD_NUM -PbuildTag=$CIRCLE_TAG -PbintrayUser=$BINTRAY_USER -PbintrayKey=$BINTRAY_KEY -PbintrayRepo=$BINTRAY_REPO -PbintrayUserOrg=$BINTRAY_USER_ORG --info --stacktrace clean bintrayUpload
+          name: Deploy release
+          command: ./gradlew --info --stacktrace publishToSonatype closeSonatypeStagingRepository
```

Comment:
to automatically release to maven there's a closeAndReleaseSonatypeStagingRepository, unless releasing it manually is intentional for this

Reply:
Yeah I'm intentionally not automatically releasing for the first release, so there's a human in the loop to catch any issues. Plan is to swap it out with `closeAndReleaseSonatypeStagingRepository` for subsequent releases. Thanks for bringing it up though.